compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/BindsMethodValidator.kt

```diff
@@ -75,28 +76,43 @@ internal class BindsMethodValidator : PrivateCodeGenerator() {
     )
 
     if (!function.parameterMatchesReturnType() && !function.receiverMatchesReturnType()) {
+      val returnType = function.returnType().asClassReference().shortName
+      val paramSuperTypes = (function.parameterSuperTypes() ?: function.receiverSuperTypes())!!
```

Comment:
May be worth tweaking the message if there's no supertypes or it's only Any, but up to you.

Reply:
Good point, updated to adjust the message when there are no supertypes, and to not list the parameter type itself in the supertype list. Will leave out special casing for Any for now as I believe we actually already filter it out for getting the supertypes.